README.md

```diff
@@ -1,3 +1,138 @@
+- Aspect Ratio: Supports 16:9, 16:10, 4:3 and custom aspect ratio
+- Scale Factor: Supports custom scale factor (e.g. 1.0, 1.5, 2.0)
+- Display Orientation: Supports manually rotating the game window during game play.
+
+## Key Mapping
+
+PlayTools provides a key mapping tool to map game actions to the keyboard, mouse or controller.
+
+Supported input devices include:
+
+- Keyboard
+- Mouse
+- Controller
+
+Input from these devices can be mapped to these in-game actions:
+
+- Button
+- Joystick (e.g. WASD)
+- Camera Control (usually by mouse)
+- Draggable Button (e.g. wheel menu selection)
+
+More mapping options such as swiping will be added in the future.
+
+## Bypassing
+
+Games designed for iOS devices may not work properly on macOS. PlayTools provides a solution to bypass these issues.
+
+### Jailbreak Bypassing
+
+Some games refuse to work on macOS because they detect the environment as a jailbroken device. PlayTools hijacks the detection process of popular jailbreak detection tools, so that the game will think it is running on a non-jailbroken device.
+
+PlayTools also allows per-game configuration for more advanced jailbreak bypassing techniques.
+
+### PlayChain
+
+PlayChain is a tool that solves key chain issues by replacing the game's key chain with a custom one. Key chain issues usually prevent the game from logging in.
+
+### Introspection Library
+
+Some games only work under debug environment. By inserting the Introspection Library, PlayTools tricks the game into thinking it is running under debug environment.
+
+# How to Use
+
+PlayTools is shipped with PlayCover, and is installed together with the game. You do not need to install PlayTools manually.
+
+PlayTools provides in-game menus mostly for key mapping setup. Other features are accessible through the PlayCover settings menu.
+
+Localization is handled in [Weblate](https://hosted.weblate.org/projects/playcover/).
+
+# How it Works
+
+PlayTools runs alongside of the game. During IPA installation, PlayTools is inserted into the game's executable. When the game is launched, PlayTools will be launched together.
+
+PlayTools uses swizzle techniques to replace framework methods and system calls with the versions provided by PlayTools. During game play, PlayTools intercepts game input events and translates them to the corresponding touch events and feed them to the game.
+
+# How to Build
+
+PlayTools is built using Xcode.
+
+## Production Build
+In release builds, the building script of PlayCover will automatically fetch the latest version of PlayTools from the official repository and build it. Generally, you do not need to build PlayTools manually.
+
+If you want to build PlayCover with a specific version of PlayTools, mostly for testing purposes, you can change the [Cartfile](https://github.com/PlayCover/PlayCover/blob/develop/Cartfile) of PlayCover to point to the specific version of PlayTools. You can specify which branch/tag of which repository to build from.
```

Comment:
I'm not sure if this is required, but is editing `Cartfile.resolved` required? Also, it could be helpful to say how to edit the Cartfile for local development instead of needing to push to Github every time.

Reply:
Theoretically editing `Cartfile` will automatically change `Cartfile.resolved`, but this process may sometimes fail. And I didn't even know you can edit the Cartfile for local development? I was manually copying the PlayTools binary all the time.

Reply:
Ok. You can edit the Cartfile to be `git file:///path/to/playtools` for local if I remember correctly (https://github.com/Carthage/Carthage/issues/1458).